### ISSUE TITLE
Eventually move away from Irrlicht

### ISSUE THREAD

**Issue by qwertychouskie** [supertuxkart/stk-code#2894]
Back in 2010, when STK 0.7 was released, Irrlicht was modern, frequently updated, etc.  It was great, it provided everything STK could need in an engine, reducing required developer manpower.

Now it's 2017.  Things have changed.  Irrlicht is all but abandoned, STK uses it's own renderer, and is getting it's own scene manager and mesh loader...  It might finally be time for STK to dump Irrlicht and use something more maintained.

Currently, STK still uses Irrlicht for:

- Scene management (SP will replace this soon though)
- Fallback rendering for older versions of OpenGL (also will be removed with SP)
- Particle system (only supports billboard particles, resulting in ugly skidsparks)
- Window+OGL context creation and management
- Input management (which is causing headaches like #2872)
- GUI (ish, it's kind of a hybrid system)

The first two won't be an issue once SP is merged.  For the rest there are a few options:

Godot 3 (currently in alpha, should be out soon):
Handles everything STK could ever need it to, in active development, large user base, supports all platforms (Windows, OS X, Linux, *BSD, Haiku, iOS, Android, BlackBerry OS, HTML5), etc.  Since Godot was open-sourced in 2014, development has happened at a breakneck pace, with no sign of slowing down.
Handles:
- Everything including the kitchen sink and VR

Potential downsides:
- Big, might take a while to integrate into STK

My opinion is this is the best option.

SDL2:
Well known, well used, supported on pretty much all platforms (including Android and Wayland).
Handles:
- Window+OGL context creation and management
- Input management
Potential downsides:
- Doesn't handle GUI or particles, additional libraries would be required for this.

GLFW doesn't support Android, so probably a no-go.

SFML is similar to SDL2, though Android support is "soon".

Obviously eliminating Irrlicht won't happen immediately, but it is probably a good idea to start discussing options now.

**Comment by deveee** [comment]
Qwerty I will eat you

2017-08-02 19:32 GMT+02:00 QwertyChouskie <notifications@github.com>:

**Comment by auriamg** [comment]
@qwertychouskie Please. Stop these kinds of post. You are a newcomer to the project, you are in a good position to make small bugfixes, etc. It is up to lead developers to decide whether an engine change is appropriate, **not** up to the newcomer. We are the ones that can evaluate a proper migration strategy, ad just posting "let's switch engines!" falls way short is not helping us.

This _will_ likely happen gradually over time, but this ticket is pointless. Please, if you want to help, start by fixing some bugs, you don't start in a project by telling people to change everything.

**Comment by qwertychouskie** [comment]
sorry was just trying to consolidate info for easy reference

**Comment by deveee** [comment]
Just to clarify it...

Since 18.02.2016 I received from you 612 emails (plus 140 emails with failed build from launchpad). In this time I received 3217 emails from all github users in general. So ~20% of all github messages come from you. This is sick!!!

Regarding to your pull requests, based on last two examples:
- In #2890 hiker had a lot of patience and explained you why you did it wrong. At least for me it's faster to remove these files myself than explaining you why it can't be merged. Especially that I have to check your pull requests carefully, because I simply don't trust you.
- In #2895 you did it wrong too and you probably have no idea why there is the 0.73 value. And I don't know why you opened this pull request, because in the first sentence I assigned it to myself. I simply didn't have possibility to test it on desktop with GL renderer and that's why I didn't push it to master yet.

In short, I'm not going to work on your pull requests, because it takes too much of time with very little benefit (if any) and I'm also tired because of your very frequent emails, so I will probably just add a rule on my mailbox to filter your messages...
